### PR TITLE
Update CreateTopicAndPublishMessages.java

### PR DESCRIPTION
**Pull request by michaelawyu**
Add the publisher error handler sample. As requested, the error handling part in the publishing quickstart sample is now removed.

### CODE REVIEW

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateTopicAndPublishMessages.java [hunk 1]
```diff
@@ -78,6 +80,54 @@ public static void publishMessages() throws Exception {
...
   }
 
+  public static void publishMessagesWithErrorHandler() throws Exception {
+    // [START pubsub_publish_error_handler]
+    ProjectTopicName topicName = ProjectTopicName.of("my-project-id", "my-topic-id");
+    Publisher publisher = null;
+
+    try {
+      // Create a publisher instance with default settings bound to the topic
+      publisher = Publisher.newBuilder(topicName).build();
```
Comment (pongad) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateTopicAndPublishMessages.java]: Should we use the try-with-resource here?

**Comment by michaelawyu** [comment]
@pongad Updated as requested, thanks!

**Comment by michaelawyu** [comment]
@pongad Sorry, my bad; I didn't realize that Publisher class hasn't implemented java.lang.AutoClosable. I will revert the update now.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateTopicAndPublishMessages.java (hunk 1 above)
Comment (michaelawyu) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateTopicAndPublishMessages.java]: It seems that Publisher hasn't implemented java.lang.AutoClosable :(

Comment (pongad) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateTopicAndPublishMessages.java]: You're right. The shutdown is longrunning so it's not closeable in the "normal" sense.

**Pull request merged**